--- a/include/charls/charls.h
+++ b/include/charls/charls.h
@@ -466,7 +466,7 @@ class jpegls_decoder final
     CHARLS_NO_DISCARD spiff_header read_spiff_header(bool& header_found) const
     {
         std::error_code ec;
-        const spiff_header header{read_spiff_header(header_found, ec)};
+        const spiff_header header(read_spiff_header(header_found, ec));
         check_jpegls_errc(static_cast<jpegls_errc>(ec.value()));
         return header;
     }
